### PR TITLE
chore: added npm cache to build

### LINKED ISSUE #716: Add cache to github actions
https://github.com/actions/cache

### PR DESCRIPTION
**Pull request by JohnstonCode**
fixes #716

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -29,7 +29,25 @@ jobs:
...
         with:
           node-version: '10.x'
-         
+
+      - name: Npm cache Posix
+        if: startsWith(matrix.os, 'ubuntu') || startsWith(matrix.os, 'macOS')
+        uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```
Comment (edgardmessias) [inline .github/workflows/main.yml]: Maybe you can use `unix-node-${{ hashFiles('**/package-lock.json') }}`

Comment (edgardmessias) [reply .github/workflows/main.yml]: To share between MAC and Ubuntu

Comment (JohnstonCode) [reply .github/workflows/main.yml]: Yeah makes sense

**Pull request merged**